types/react-table/Readme.md

```diff
@@ -140,7 +140,7 @@ declare module 'react-table' {
       UseRowStateState<D>,
       UseSortByState<D> {}
 
-  export interface Column<D extends object = {}>
+  export interface ColumnInterface<D extends object = {}>
```

Comment:
I think that we should add a note to the top of the readme saying that if you were previously using version 7.0.13 or earlier of @types/react-table, that you should pay particular attention to the change from Column to ColumnInterface.

I foresee that being a pain point for most users and there's no obvious way to tell users why their code is breaking.

Maybe we should add a changelog to the types file?

Thoughts?

Reply:
Can we bump to a minor version, rather than just a patch?

Reply:
To clarify: @types/react-table - 7.1.0. I think this is a big enough change that it warrants a minor version.

Reply:
I don't think that we can do that.  The DT pattern is that major and minor match the major and minor of the package that they are providing types for.  So, we go to 7.1, when react-table goes to 7.1

I'm having a discussion on the typescript discord about this right now, the gist is that yes, patches break things, and that DT doesn't follow semver.

Reply:
Well if DT doesn't follow semver, then I think we need to paste this change all over the readme. I can see this being a bit of an issue because people who don't fix versions are going to have a bit of a surprise the next time they load packages.

Reply:
It is unfortunate that the types can't have their own semver, but oh well. I have added a changelog to the Readme mentioning this PR, and a mention of changelog existence in the `index.d.ts`.

Hopefully, this would be enough.

Reply:
TIL
> DT doesn't follow semver.
😢

Reply:
I learned this by hitting the breaking changes, finding this PR, and this comment thread. FYI, I am not the person responsible for the code with the breaking changes, or even the initial use of react-table. If there's any real world example of the obvious problem this thread describes, that was it. So far, just locked to `7.0.13` because the changes are not straight forward.